server/channels/app/plugin_api.go

```diff
@@ -1531,6 +1531,19 @@ func (api *PluginAPI) CreatePropertyField(field *model.PropertyField) (*model.Pr
 		return nil, fmt.Errorf("invalid input: property field parameter is required")
 	}
 
+	cpaGroupID, err := api.app.CpaGroupID()
+	if err != nil {
+		return nil, model.NewAppError("CreatePropertyField", "app.custom_profile_attributes.cpa_group_id.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	// Include plugin ID in CPA fields
+	if field.GroupID == cpaGroupID {
+		if field.Attrs == nil {
+			field.Attrs = make(model.StringInterface)
+		}
+		field.Attrs[model.CustomProfileAttributesPropertyAttrsSourcePluginID] = api.manifest.Id
```

Comment:
Will it be the plugin responsibility to set `protected` when they define their attribute through this plugin api?